File: frontend/src/app/components/doctores/doctor.ts
export class Doctor{
    constructor(
        public id:string,
        public nombre:string,
        public apellido:string,
        public especialidad:string,
        public hospital:string,
        public codigoProfesional:string
    ){}
}

export class Paciente{
    constructor(
        public id:string,
        public nombre:string,
        public apellido:string,
        public telefono:string
    ){}
}

File: backend/Nexos.Domain/Entities/Doctor.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Nexos.Domain.Entities
{
    public class Doctor : Entity
    {
        [Required]
        public string Nombre { get; set; }

        [Required]
        public string Apellido { get; set; }

        [Required]
        public string CodigoProfesional { get; set; }

        [Required]
        public string Especialidad { get; set; }

        [Required]
        public string Hospital { get; set; }

        public virtual ICollection<Paciente> Pacientes { get; set; }
    }
}
